### PR TITLE
Story 114.2: Implement Disconnected-Header Layout and Scrollbar Ownership Fix

### PR DESCRIPTION
**Pull request by DaveMBush**
## Summary
- Move disconnected header markup into sibling header viewport so header stays fixed while body scrolls.
- Keep ".dms-outer-scroller" as vertical owner, move horizontal ownership to ".dms-table-body", and sync header/body geometry without restoring sticky headers.
- Update focused regression coverage and worktree-aware affected-project quality gate logic for touched projects.

## Testing
- pnpm exec playwright test src/base-table-layout-regression.spec.ts src/base-table-two-region-regression.spec.ts --config=apps/dms-material-e2e/playwright.config.ts --project=chromium
- pnpm exec playwright test src/accessibility.spec.ts --config=apps/dms-material-e2e/playwright.config.ts --project=chromium --grep "should have proper table structure on universe page"
- pnpm exec nx run dms-material:lint
- pnpm exec nx run dms-material-e2e:lint
- pnpm exec nx run dms-material:test
- CI=1 pnpm all

Closes #1329

## Summary by CodeRabbit

* **New Features**
  * Header–body scroll synchronization and header interaction utilities for the base table
  * Conditional Storybook/e2e startup and dynamic test server routing for local worktrees

* **Bug Fixes**
  * Fixed disconnected header layout and horizontal scroll forwarding/sync issues
  * Resolved table scroll/selection interaction edge cases and keyboard focus behaviors

* **Refactor**
  * Separated header and body scroll ownership in table layout and templates
  * Simplified table initialization wiring

* **Tests**
  * Expanded E2E and unit tests for scrolling, layout regressions, accessibility, and data workflows

* **Chores**
  * Replaced monolithic QA script with an affected-project quality-runner and added proxy/testing env wiring

### CODE REVIEW

File: apps/dms-material-e2e/playwright.config.ts [hunk 1]
```diff
@@ -20,28 +27,60 @@ if (process.env.WSL_DISTRO_NAME && !process.env.CI) {
...
+const apiBaseUrl =
+  process.env['E2E_API_BASE_URL'] ??
+  `http://localhost:${String(serverPort)}/api`;
+const reuseExistingServer = !isGitWorktree;
+const hasExplicitTestPath = process.argv.some(
+  (argument) => argument.endsWith('.spec.ts') || argument.includes('/src/')
+);
+const requiresStorybook =
+  process.argv.some((argument) => argument.includes('storybook')) ||
+  !hasExplicitTestPath;
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/dms-material-e2e/playwright.config.ts:78]: Verified current regression specs. `base-table-layout-regression.spec.ts` and `base-table-two-region-regression.spec.ts` log in and navigate application routes such as `/global/universe`; they do not reference `STORYBOOK_BASE_URL` or Storybook helpers. Disabling Storybook for those explicit spec runs is intentional here, so I am leaving the config unchanged.
